### ISSUE TITLE
kube-dns deployment occasionally fails with juju

### ISSUE THREAD

**Issue by Cynerva** [kubernetes/kubernetes#40386]
**Kubernetes version** (use `kubectl version`):
1.5.2

**Environment**:
- **Cloud provider or hardware configuration**: LXD
- **OS** (e.g. from /etc/os-release): Ubuntu 16.04
- **Kernel** (e.g. `uname -a`): 4.4.0-57-generic #78-Ubuntu
- **Install tools**: Juju
- **Others**:

**What happened**:

The kubernetes-master charm entered a `hook failed: "update-status"` state with the following stacktrace:
```
The Service "kube-dns" is invalid: spec.clusterIP: Invalid value: "10.152.183.10": provided IP is already allocated
2017-01-24 20:52:16 INFO update-status Traceback (most recent call last):
2017-01-24 20:52:16 INFO update-status   File "/var/lib/juju/agents/unit-kubernetes-master-0/charm/hooks/update-status", line 19, in <module>
2017-01-24 20:52:16 INFO update-status     main()
2017-01-24 20:52:16 INFO update-status   File "/usr/local/lib/python3.5/dist-packages/charms/reactive/__init__.py", line 78, in main
2017-01-24 20:52:16 INFO update-status     bus.dispatch()
2017-01-24 20:52:16 INFO update-status   File "/usr/local/lib/python3.5/dist-packages/charms/reactive/bus.py", line 434, in dispatch
2017-01-24 20:52:16 INFO update-status     _invoke(other_handlers)
2017-01-24 20:52:16 INFO update-status   File "/usr/local/lib/python3.5/dist-packages/charms/reactive/bus.py", line 417, in _invoke
2017-01-24 20:52:16 INFO update-status     handler.invoke()
2017-01-24 20:52:16 INFO update-status   File "/usr/local/lib/python3.5/dist-packages/charms/reactive/bus.py", line 291, in invoke
2017-01-24 20:52:16 INFO update-status     self._action(*args)
2017-01-24 20:52:16 INFO update-status   File "/var/lib/juju/agents/unit-kubernetes-master-0/charm/reactive/kubernetes_master.py", line 313, in start_kube_dns
2017-01-24 20:52:16 INFO update-status     create_addon('kubedns-svc.yaml', context)
2017-01-24 20:52:16 INFO update-status   File "/var/lib/juju/agents/unit-kubernetes-master-0/charm/reactive/kubernetes_master.py", line 431, in create_addon
2017-01-24 20:52:16 INFO update-status     check_call(cmd)
2017-01-24 20:52:16 INFO update-status   File "/usr/lib/python3.5/subprocess.py", line 581, in check_call
2017-01-24 20:52:16 INFO update-status     raise CalledProcessError(retcode, cmd)
2017-01-24 20:52:16 INFO update-status subprocess.CalledProcessError: Command '['kubectl', 'apply', '-f', '/etc/kubernetes/addons/kubedns-svc.yaml']' returned non-zero exit status 1
```

The 10.152.183.10 IP was already claimed by default-http-backend:
```
$ kubectl get svc default-http-backend
NAME                   CLUSTER-IP      EXTERNAL-IP   PORT(S)   AGE
default-http-backend   10.152.183.10   <none>        80/TCP    32m
```

**What you expected to happen**:

default-http-backend should not deploy until after kube-dns has had a chance to reserve its dedicated IP

**How to reproduce it** (as minimally and precisely as possible):

It'll be hard to reproduce this reliably since the default-http-backend service IP is chosen randomly. However, this might occur sporadically when deploying with juju.

**Comment by mbruzek** [comment]
It looks like we are pinning kube-dns to *.*.*.10 and the default-http-backend already claimed it. 

What are other providers doing? Are they pinning dns to an IP address?

**Comment by Cynerva** [comment]
@calebamiles I don't think this needs to be fixed for 1.6. It's a pretty rare case and only applies to deployments with Juju.

**Comment by grodrigues3** [comment]
removing the 1.6 milestone as per @Cynerva
